app/2.0/docs/tools/build-for-production.md

```diff
@@ -0,0 +1,222 @@
+
+By default, the Polymer build tools rejoin JavaScript and HTML after processing.
+
+To split your code from HTML in order to deploy into an environment that requires this, you can use [Crisper](https://github.com/PolymerLabs/crisper).
+  
+### Minifying HTML, JavaScript and CSS {#minifying}
+
+The Polymer build tools provide options to minify HTML, JavaScript and CSS by stripping whitespace and comments. You can use these on the command line, or configure them in `polymer.json`. See the [polymer.json specification](polymer-json) for more details.
+
+**Function**|**CLI flag**|**Entry in polymer.json**
+--- | --- | ---
+Minify inlined and external JavaScript | `--js-minify` | `"js": {"minify": true}` 
+Minify inlined and external CSS | `--css-minify` | `"css": {"minify": true}` 
+Minify HTML | `--html-minify` | `"html": {"minify": true}` 
+
+Example: Build a project. Use CLI options to minify JavaScript, CSS and HTML {.caption}
+
+```bash
+polymer build --js-minify --css-minify --html-minify 
+```
+
+Example: A build object in `polymer.json` that minifies JavaScript, CSS and HTML {.caption}
+
+```
+"builds": [{
+  "js": {"minify": true},
+  "css": {"minify": true},
+  "html": {"minify": true}
+}]
+```
+
+### Compiling ES6 to ES5 {#compiling}
+
+Polymer 2.x and its native elements are written using ES6, allowing class definitions, inheritance and modular code. Support for ES6 is required in order for a browser to implement the [custom elements](https://developers.google.com/web/fundamentals/getting-started/primers/customelements) specification.
+
+Because Polymer 2.0 uses ES6 and HTML Custom Elements, it is always best to serve ES6 to browsers with full ES6 support.
+
+These browsers fully support ES6:
+
+* Chrome or Chromium version 49 or later.
+* Opera 36 or later.
+* Safari or Mobile Safari 10 or later.
+* Edge 15.15063 or later.
+* Firefox 51 or later.
+
+You should serve compiled ES5 only to older browsers that don't support ES6.
+
+If you need to statically host your code and serve a single version to all browsers, however, you should compile all code to ES5. In this case, you can include a shim - a lightweight polyfill that lets compiled ES5 work on browsers that support native custom elements.
+  
+```bash
+polymer build --js-compile 
+```
+This flag adds the `custom-elements-es5-adapter.js` adapter for running ES5 code on browsers that support ES6.
+
+If you’re unsure what the best strategy is for your project, here’s a quick overview:
+
+|   | Easiest for cross-browser support  | Most optimal for WC v1 performance  |
+|---|-------|------|
+| **Server** | Any server works, including static ones | Differential serving required |
+| **Deployed Code** | ES5 transpiled | ES6|
+| **Polyfill Loader** | custom-elements-es5-adapter.js | webcomponents-loader.js|
```

Comment:
This table still isn't right. And I think it's going to be hard to record in exactly this format.

Can probably shorten "Most optimal ..." to just "Best performance." This should perform better on all browsers that support ES6, not just ones with native CE (because the native ES6 will in most cases run faster than the transpiled version).

First row, "differential serving => dynamic serving"?
For deployed code, the first column should be "Single build, ES5 transpiled" (Or just ES5 like you have now), while the other should say, "Two builds, ES5 and ES6"

Polyfill loader: either drop this row or rewrite it as "Polyfills" ... Column one should be custom-elements-es5-adapter.js + webcomponents-loader.js and column two stays the same. (Really, it is either webcomponents-loader.js or webcomponents-lite.js. I suspect we're going to be pushing the latter as more production-friendly.